core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,7 +64,23 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // Evaluate partitions in parallel. Partitions of each rdd will be cached by the `partitions`
+  // val in `RDD`.
+  private[spark] lazy val parallelPartitionEval: Boolean = {
+    val threshold = conf.getInt("spark.rdd.parallelListingThreshold", 10)
+    if (threshold > 0 && rdds.length > threshold) {
+      val parArray = rdds.toParArray
+      parArray.tasksupport = new ForkJoinTaskSupport(
+        new ForkJoinPool(math.min(threshold, 8)))
```

Comment:
In this case, it will always be 8 threads by default. How about 

```scala
  private[spark] lazy val parallelPartitionEval: Boolean = {
    val parallelism = conf.getInt("spark.rdd.listingParallelism", 16)
    val threshold = 8
    if (rdds.length > threshold) {
      val parArray = rdds.toParArray
      parArray.tasksupport = new ForkJoinTaskSupport(
        new ForkJoinPool(math.min(parallelism, rdds.length)))
      parArray.foreach(_.partitions)
      true
    } else {
      false
    }
  }
```

Thanks.